sycl/include/CL/sycl/detail/pi.h

```diff
@@ -855,6 +855,17 @@ pi_result piEnqueueMemUnmap(
   const pi_event * event_wait_list,
   pi_event *       event);
 
+pi_result piGetDeviceFunctionPointer(
+  pi_device        device,
+  pi_program       program,
+  const char     * function_name,
+  pi_uint64      * function_pointer_ret);
+
+pi_result piGetExtensionFunctionAddressForPlatform(
```

Comment:
Please do not add this opaque interface

Reply:
What if I refactor it into a `piGetExtensionFunctionAddress` helper which accepts `pi_device`, automatically extracts corresponding `pi_platform` and then queries desired function pointer?

Right now this extract platform from device thing is performed at least in two places: in `piGetDeviceFunctionPointer` and in `OCL(piProgramCreate)`

Reply:
> helper which accepts `pi_device`, automatically extracts corresponding `pi_platform` and then queries desired function pointer?

We do not need an interface in PI that returns some function by its name (how extensions are implemented in OpenCL). Of course, you can create a helper internal to the OpenCL PI plugin, though.